src/sinks/gcp/pubsub.rs

```diff
@@ -118,13 +121,13 @@ struct PubsubSink {
 impl PubsubSink {
     async fn from_config(config: &PubsubConfig) -> crate::Result<Self> {
         // We only need to load the credentials if we are not targeting an emulator.
-        let creds = match config.emulator_host {
-            None => config.auth.make_credentials(Scope::PubSub).await?,
-            Some(_) => None,
+        let creds = match &config.skip_authentication {
+            true => None,
+            _ => config.auth.make_credentials(Scope::PubSub).await?,
```

Comment:
This doesn't need a reference (`&config.skip_authentication`) and I think clippy will want you to write this as an if/else block instead.

Reply:
rust doesn't have a ternary operator?  whats the accepted convention to use here?

Reply:
No, Rust has no ternary operator, but `if else` is an expression. This is what I would expect clippy to suggest:
```suggestion
        let creds = if config.skip_authentication {
            None
        } else {
            config.auth.make_credentials(Scope::PubSub).await?
```

Reply:
thanks!